db/version_set.cc

```diff
@@ -3130,6 +3130,10 @@ bool Version::MaybeInitializeFileMetaData(const ReadOptions& read_options,
   file_meta->raw_value_size = tp->raw_value_size;
   file_meta->raw_key_size = tp->raw_key_size;
   file_meta->num_range_deletions = tp->num_range_deletions;
+  // Ensure new invariants on old files
+  file_meta->num_deletions =
+      std::max(tp->num_deletions, tp->num_range_deletions);
+  file_meta->num_entries = std::max(tp->num_entries, tp->num_deletions);
```

Comment:
I think `tp->num_entries >= tp->num_deletions` even for old files. Did you mean `std::max(tp->num_entries, tp->num_range_deletions);`?

Reply:
Just being slightly more proactive against unexpected serialized data breaking an invariant.